tests/test_trusted_metadata_set.py

```diff
@@ -110,13 +137,232 @@ def test_update_with_invalid_json(self):
+        trusted_set = TrustedMetadataSet(self.metadata["root"])
+        # call root_update_finished when trusted root has expired
+        expired_datetime = datetime.strptime(
+            "1970-01-01T00:00:00Z", "%Y-%m-%dT%H:%M:%SZ"
+        )
+        trusted_set.root.signed.expires = expired_datetime
+        with self.assertRaises(exceptions.ExpiredMetadataError):
+            trusted_set.root_update_finished()
+
+    def _sign_modified_obj(
+        self,
+        role:str,
+        metadata_obj: Metadata
+    ) -> Dict[str, Any]:
+        key_dict = self.keystore[role]
+        sslib_signer = SSlibSigner(key_dict)
+        signature = metadata_obj.sign(sslib_signer)
+        return signature.to_dict()
+
+    def test_update_timestamp(self):
+        trusted_set = TrustedMetadataSet(self.metadata["root"])
+        trusted_set.root_update_finished()
+        trusted_set.update_timestamp(self.metadata["timestamp"])
+        # new_timestamp.version < trusted_timestamp.version
+        trusted_set.timestamp.signed.version = 2
+        with self.assertRaises(exceptions.ReplayedMetadataError):
+            trusted_set.update_timestamp(self.metadata["timestamp"])
+        trusted_set.timestamp.signed.version = 1
+
+        # new_timestamp.snapshot.version < trusted_timestamp.snapshot.version
+        trusted_set.timestamp.signed.meta["snapshot.json"].version = 2
+        with self.assertRaises(exceptions.ReplayedMetadataError):
+            trusted_set.update_timestamp(self.metadata["timestamp"])
+        trusted_set.timestamp.signed.meta["snapshot.json"].version = 1
+
+        # new_timestamp has expired
+        timestamp = Metadata.from_bytes(self.metadata["timestamp"])
+        timestamp.signed.expires = datetime.strptime(
+            "1970-01-01T00:00:00Z", "%Y-%m-%dT%H:%M:%SZ"
+        )
+        self._sign_modified_obj("timestamp", timestamp)
+        new_timestamp_byte_data = json.dumps(timestamp.to_dict()).encode()
+        with self.assertRaises(exceptions.ExpiredMetadataError):
+            trusted_set.update_timestamp(new_timestamp_byte_data)
+
+    def _calculate_modified_hashes(
+        self, true_hashes,
+        data: bytes
+    ) -> Dict[str, str]:
+        modified_hashes = {}
+        # Calculate correct hashes to pass the first check
+        for algo in true_hashes.keys():
+            digest_object = sslib_hash.digest(algo)
+            digest_object.update(data)
+            observed_hash = digest_object.hexdigest()
+            modified_hashes[algo] = observed_hash
+        return modified_hashes
+
+    def test_update_snapshot(self):
+        trusted_set = TrustedMetadataSet(self.metadata["root"])
+        trusted_set.root_update_finished()
+        trusted_set.update_timestamp(self.metadata["timestamp"])
+        # cannot update snapshot after targets update completes or targets != None
+        targets_obj = Metadata.from_bytes(self.metadata["targets"])
+        trusted_set._trusted_set["targets"] = targets_obj
```

Comment:
```suggestion
        trusted_set.targets = targets_obj
```

Reply:
At the time of the assignment targets is `None` and that's why I assign it that way.